### DISCUSSION TITLE
usb adapter - suggestion

### DISCUSSION THREAD

**Discussion by heapxor** [Koenkk/zigbee2mqtt#12004] [Devices]
hello,
on the webpage https://www.zigbee2mqtt.io/guide/adapters/#recommended there are the following adapters suggested
But is it possible to say which one is the best stable/ supported etc? Or based on what do i have to decide? thanks!

USB connected (easiest)

    Electrolama zig-a-zig-ah! (zzh!)
    Slaesh's CC2652RB stick
    Tube's CC2652P2 USB Coordinator
    Egony Stick V4
    ZigStar Stick v4
    CircuitSetup's CC2652P2 USB Coordinator
    SMARTLIGHT CC2652P Zigbee USB Adapter SLZB-02
    SONOFF Zigbee 3.0 USB Dongle Plus
    Vision CC2652 dongle

**Comment by robertalexa** [comment]
For quite some time, if my memory serves me right, the documentation said "recommended" next to zzh! :)

Personally I am running on zzh! and have never ever had any problems, i have a network of 40ish devices, with a few routers as well. Devices react instantly, I have never had freezes, "wtf"s, nothing. Firmware comes from https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator

I did not test any of the other recommended devices, so I can't give my opinion on them - so don't think that i say "no don't use them", i just don't have 1st hand experience with them

Hope this helps
